tidb-in-kubernetes: add tidb-backup configuration reference (#1423)
pingcap/docs-cn: tidb-in-kubernetes/tidb-backup-configuration.md

---
title: TiDB Backup 配置参数
category: reference
---

# TiDB Backup 配置参数

> `TiDB-Backup` 是一个用于 TiDB 集群备份和恢复的 Helm Chart, 这篇文章将详细介绍 `TiDB-Backup` 的可配置参数。参考[备份与恢复](/dev/how-to/maintain/backup-and-restore.md)了解如何使用 `TiDB-Backup` 进行集群备份恢复。

Reviewer: ```suggestion
- Please keep the name consistent. Let's use "TiDB Backup"?
- Use Chinese punctuation where necessary.

TiDB Backup 是一个用于 TiDB 集群备份和恢复的 Helm Chart。本文将详细介绍 TiDB Backup 的可配置参数。要了解如何使用 TiDB Backup 进行集群备份恢复，可参考[备份与恢复](/dev/how-to/maintain/backup-and-restore.md)。
```

> 阅读本文档前，请参考 [Helm 使用说明](/TODO/to-helm-guide) 来了解如何修改这些配置项。 

Reviewer: This is a broken link.

Author: Yes, the "Helm User Guide" has not completed yet

Reviewer: @aylei Let's remove the link now and add it back later when the file is ready?

## `mode`

+ 运行模式

> + 默认： "backup"

Reviewer: **DO NOT** add a space before or after a Chinese punctuation mark.

```suggestion
+ 默认："backup"
```

> + 可选值为 `backup`(备份集群数据) 和 `restore`(恢复集群数据)

Reviewer: ```suggestion
+ 可选值为 `backup`（备份集群数据）和 `restore`（恢复集群数据）
```

## `clusterName`

+ 目标集群名字

> + 默认： "demo"

Reviewer: ```suggestion
+ 默认："demo"
```

Author: Got it👌

+ 指定要从哪个集群进行备份或将数据恢复到哪个集群中

## `name`

+ 备份名

> + 默认值： "fullbackup-${date}"，`date` 是备份的开始时间，精确到分钟

Reviewer: ```suggestion
+ 默认值："fullbackup-${date}"，`date` 是备份的开始时间，精确到分钟
```

+ 备份名用于区分不同的备份数据

## `secretName`

+ 访问目标集群时使用的凭据

> + 默认： "backup-secret"

Reviewer: ```suggestion
+ 默认："backup-secret"
```

+ 该 Kubernetes Secret 中需要存储目标集群的登录用户名和密码，你可以通过以下命令来创建这个 Secret：

>     ```shell

Reviewer: ```suggestion
    
    ```shell
```

    $ kubectl create secret generic backup-secret -n ${namespace} --from-literal=user=root --from-literal=password=<password>
    ```

## `storage.className`

+ Kubernetes StorageClass 

> + 默认： "local-storage"

Reviewer: ```suggestion
+ 默认："local-storage"
```

> + 备份任务需要绑定一个持久卷来永久或临时存储备份数据，`StorageClass` 用于声明持久卷使用的存储类型,，你需要确保该 `StorageClass` 在 Kubernetes 集群中存在。 

Reviewer: Do you mean "Persistent Volume, PV" for "持久卷"? If yes, it would be better to provide the English name.

```suggestion
+ 备份任务需要绑定一个持久卷 (Persistent Volume, PV) 来永久或临时存储备份数据，`StorageClass` 用于声明持久卷使用的存储类型，需要确保该 `StorageClass` 在 Kubernetes 集群中存在。 
```

## `storage.size`

+ 持久卷的空间大小
+ 默认："100Gi"

## `backupOptions`

+ 备份参数

> + 默认： "--chunk-filesize=100"

Reviewer: ```suggestion
+ 默认："--chunk-filesize=100"
```

+ 为备份数据时使用的 [mydumper](/dev/reference/tools/mydumper.md) 指定额外的运行参数

## `restoreOptions`

+ 恢复参数

> + 默认： "-t 16"

Reviewer: ```suggestion
+ 默认："-t 16"
```

> + 为恢复数据时使用的 [loader](/dev/reference/tools/loader.md) 指定额外的运行参数 

Reviewer: ```suggestion
+ 为恢复数据时使用的 [Loader](/dev/reference/tools/loader.md) 指定额外的运行参数 
```

## `gcp.bucket`

+ Google Cloud Storage 的 bucket 名字
+ 默认：""
+ 用于存储备份数据到 Google Cloud Storage 上

> **注意：**
>
> 一旦设置了 `gcp` 下的任何参数，备份和恢复就会使用 Google Cloud Storage 进行数据存储。也就是说，假如要设置 `gcp` 下的参数，就需要保证所有 `gcp` 相关参数都得到正确配置，否则会造成备份恢复失败。

## `gcp.secretName`

+ 访问 GCP 时使用的凭据
+ 默认：""
+ 该 Kubernetes Secret 中需要存储 GCP 的 Service Account 凭据，你可以参考 [Google Cloud Documentation](https://cloud.google.com/docs/authentication/production#obtaining_and_providing_service_account_credentials_manually) 来下载凭据文件，然后通过下面的命令创建 Secret：

    ```shell
    $ kubectl create secret generic gcp-backup-secret -n ${namespace} --from-file=./credentials.json
    ```

## `ceph.endpoint`

+ Ceph 对象存储的 Endpoint
+ 默认：""
+ 用于访问 Ceph 对象存储

> **注意：**
>
> 一旦设置了 `ceph` 下的任何参数，备份和恢复就会使用 Ceph 对象存储进行数据存储。也就是说，假如要设置 `ceph` 下的参数，就需要保证所有 `ceph` 相关参数都得到正确配置，否则会造成备份恢复失败。

## `ceph.bucket`

> + Ceph 对象存储的 buckt 名字

Reviewer: ```suggestion
+ Ceph 对象存储的 bucket 名字
```

+ 默认：""
+ 用于存储数据到 Ceph 对象存储上

## `ceph.secretName`

+ 访问 Ceph 对象存储时使用的凭据
+ 默认：""

> +_该 Kubernetes Secret 中需要存储访问 Ceph 时使用的 `access_key` 和 `secret_key`。你可以使用下面的命令来创建这个 Secret：

Reviewer 2: ```suggestion
+ 该 Kubernetes Secret 中需要存储访问 Ceph 时使用的 `access_key` 和 `secret_key`。你可以使用下面的命令来创建这个 Secret：
```

---
title: TiDB Backup 配置参数
category: reference
---

# TiDB Backup 配置参数

TiDB-Backup 是一个用于 TiDB 集群备份和恢复的 Helm Chart。本文详细介绍了 TiDB-Backup 的可配置参数。了解如何使用 TiDB-Backup 进行集群备份恢复，请参考[备份与恢复](/dev/how-to/maintain/backup-and-restore.md)。

## `mode`

+ 运行模式
+ 默认："backup"
+ 可选值为 `backup`（备份集群数据）和 `restore`（恢复集群数据）

## `clusterName`

+ 目标集群名字
+ 默认："demo"
+ 指定要从哪个集群进行备份或将数据恢复到哪个集群中

## `name`

+ 备份名
+ 默认值："fullbackup-${date}"，`date` 是备份的开始时间，精确到分钟
+ 备份名用于区分不同的备份数据

## `secretName`

+ 访问目标集群时使用的凭据
+ 默认："backup-secret"
+ 该 Kubernetes Secret 中需要存储目标集群的登录用户名和密码，你可以通过以下命令来创建这个 Secret：

    {{< copyable "shell-regular" >}}
    
    ```shell
    kubectl create secret generic backup-secret -n ${namespace} --from-literal=user=root --from-literal=password=<password>
    ```

## `storage.className`

> + Kubernetes StorageClass

Reviewer 3: ```suggestion
+ Kubernetes StorageClass
```
是 Kubernetes Storage Class 吗

---
title: TiDB Backup 配置参数
category: reference
---

# TiDB Backup 配置参数

TiDB-Backup 是一个用于 TiDB 集群备份和恢复的 Helm Chart。本文详细介绍了 TiDB-Backup 的可配置参数。了解如何使用 TiDB-Backup 进行集群备份恢复，请参考[备份与恢复](/dev/how-to/maintain/backup-and-restore.md)。

> 阅读本文档前，请参考 Helm 使用说明来了解如何修改这些配置项。 

Reviewer 3: Can you add a link for "Helm 使用说明"?

Author: I've just removed the dead link, let me remove this line before the helm documentation complete😅

Reviewer 3: Sure haha
